daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala

```diff
@@ -1550,17 +1550,20 @@ private[lf] final class Compiler(
   private[this] def compileCreate(
       tmplId: Identifier,
       tmpl: Template,
-  ): (SDefinitionRef, SDefinition) =
+  ): (SDefinitionRef, SDefinition) = {
+    val precondsArray =
+      (Iterator(tmpl.precond) ++ (tmpl.implements.map(impl => impl._2.precond))).to(ImmArray)
+    val preconds = ECons(TBuiltin(BTBool), precondsArray, ENil(TBuiltin(BTBool)))
     // Translates 'create Foo with <params>' into:
     // CreateDefRef(tmplId) = \ <tmplArg> <token> ->
-    //   let _ = $checkPreconf(tmplId)(<tmplArg> [tmpl.precond]
+    //   let _ = $checkPrecond(tmplId)(<tmplArg> [tmpl.precond ++ [precond | precond <- tmpl.implements]]
     //   in $create <tmplArg> [tmpl.agreementText] [tmpl.signatories] [tmpl.observers] [tmpl.key]
     topLevelFunction2(CreateDefRef(tmplId)) { (tmplArgPos, _, _env) =>
       val env = _env.bindExprVar(tmpl.param, tmplArgPos)
       // We check precondition in a separated builtin to prevent
       // further evaluation of agreement, signatories, observers and key
       // in case of failed precondition.
-      let(env, SBCheckPrecond(tmplId)(env.toSEVar(tmplArgPos), compile(env, tmpl.precond))) {
+      let(env, SBCheckPrecond(tmplId)(env.toSEVar(tmplArgPos), compile(env, preconds))) {
```

Comment:
One thoughts (no need to tackle them in this PR): 

It may be useful to know which precondition fails: It is the precondition of the template definition or one of the interface that fail. In this case a compilation as follow may be more useful: 
```
  let _ = $checkPreconf(tmplId)(<tmplArg> [tmpl.precond]
  let _ = $checkPreconf(iface_1.id)(<tmplArg> [iface_1.precond]
  ...
  let _ = $checkPreconf(ifaceId_n.id)(<tmplArg> [iface_n.precond]```
```